scheduler/test/cook/test/mesos/monitor.clj

```diff
@@ -0,0 +1,194 @@
+    (is (= 1 (counter ["total" "users"])))
+    (is (= 1 (counter ["starved" "users"])))
+    (is (= 0 (counter ["hungry" "users"])))
+    (is (= 0 (counter ["satisfied" "users"])))
+
+    (run-job job2)
+    (monitor/set-stats-counters! (db conn) stats-atom)
+    (is (= 1 (counter ["running" "all" "jobs"])))
+    (is (= 2 (counter ["running" "all" "cpus"])))
+    (is (= 256 (counter ["running" "all" "mem"])))
+    (is (= 1 (counter ["running" "bob" "jobs"])))
+    (is (= 2 (counter ["running" "bob" "cpus"])))
+    (is (= 256 (counter ["running" "bob" "mem"])))
+    (is (= 0 (counter ["waiting" "all" "jobs"])))
+    (is (= 0 (counter ["waiting" "all" "cpus"])))
+    (is (= 0 (counter ["waiting" "all" "mem"])))
+    (is (= 0 (counter ["waiting" "alice" "jobs"])))
+    (is (= 0 (counter ["waiting" "alice" "cpus"])))
+    (is (= 0 (counter ["waiting" "alice" "mem"])))
+    (is (= 0 (counter ["waiting" "bob" "jobs"])))
+    (is (= 0 (counter ["waiting" "bob" "cpus"])))
+    (is (= 0 (counter ["waiting" "bob" "mem"])))
+    (is (= 0 (counter ["starved" "all" "jobs"])))
+    (is (= 0 (counter ["starved" "all" "cpus"])))
+    (is (= 0 (counter ["starved" "all" "mem"])))
+    (is (= 0 (counter ["starved" "alice" "jobs"])))
+    (is (= 0 (counter ["starved" "alice" "cpus"])))
+    (is (= 0 (counter ["starved" "alice" "mem"])))
+    (is (= 0 (counter ["starved" "bob" "jobs"])))
+    (is (= 0 (counter ["starved" "bob" "cpus"])))
+    (is (= 0 (counter ["starved" "bob" "mem"])))
+    (is (= 1 (counter ["total" "users"])))
+    (is (= 0 (counter ["starved" "users"])))
+    (is (= 0 (counter ["hungry" "users"])))
+    (is (= 1 (counter ["satisfied" "users"])))
+
+    (mesos/kill-job conn [(:uuid job2)])
+    (monitor/set-stats-counters! (db conn) stats-atom)
+    (is (= 0 (counter ["running" "all" "jobs"])))
+    (is (= 0 (counter ["running" "all" "cpus"])))
+    (is (= 0 (counter ["running" "all" "mem"])))
+    (is (= 0 (counter ["waiting" "all" "jobs"])))
+    (is (= 0 (counter ["waiting" "all" "cpus"])))
+    (is (= 0 (counter ["waiting" "all" "mem"])))
+    (is (= 0 (counter ["waiting" "alice" "jobs"])))
+    (is (= 0 (counter ["waiting" "alice" "cpus"])))
+    (is (= 0 (counter ["waiting" "alice" "mem"])))
+    (is (= 0 (counter ["waiting" "bob" "jobs"])))
+    (is (= 0 (counter ["waiting" "bob" "cpus"])))
+    (is (= 0 (counter ["waiting" "bob" "mem"])))
+    (is (= 0 (counter ["starved" "all" "jobs"])))
+    (is (= 0 (counter ["starved" "all" "cpus"])))
+    (is (= 0 (counter ["starved" "all" "mem"])))
+    (is (= 0 (counter ["starved" "alice" "jobs"])))
+    (is (= 0 (counter ["starved" "alice" "cpus"])))
+    (is (= 0 (counter ["starved" "alice" "mem"])))
+    (is (= 0 (counter ["starved" "bob" "jobs"])))
+    (is (= 0 (counter ["starved" "bob" "cpus"])))
+    (is (= 0 (counter ["starved" "bob" "mem"])))
+    (is (= 0 (counter ["total" "users"])))
+    (is (= 0 (counter ["starved" "users"])))
+    (is (= 0 (counter ["hungry" "users"])))
```

Comment:
Can we add a case where `hungry-users` is non-zero and then transitions to zero after the user's task starts running?

Reply:
Done